packages/tasit-action/src/TransactionSubscription.js

```diff
@@ -11,19 +11,46 @@ export class TransactionSubscription extends Subscription {
   #txConfirmations;
   #timeout;
   #lastConfirmationTime;
-
-  constructor(txPromise, provider) {
+  #contract;
+
+  // Contract passing itself to here, enables
+  // TransactionSubscription calling this.#contract._emitErrorEvent
+  // to emit error to the contract catch-all error listener
+  // As that's is the only function called for now, could be better
+  // pass that function as a callback here?
+  constructor(txPromise, provider, contract) {
     // Provider implements EventEmitter API and it's enough
     //  to handle with transactions events
     super(provider);
 
     const { events } = config;
     const { timeout } = events;
 
+    this.#txPromise = txPromise.then(
+      tx => {
+        return tx;
+      },
+      error => {
+        this._emitErrorEvent(
+          new Error(`Action with error: ${error.message}`),
+          null
+        );
+      }
+    );
+
     this.#timeout = timeout;
-    this.#txPromise = txPromise;
     this.#provider = provider;
     this.#txConfirmations = 0;
+    this.#contract = contract;
+  }
+
+  // TODO: Make protected
+  // Arrow functions in class properties won't be in the prototype and we can't call them with super
```

Comment:
```suggestion

  // Arrow functions in class properties won't be in the prototype and we can't call them with super
```

Reply:
Adding this space because the idea of making it protected and the rationale for it being a non-arrow function are two separate thoughts